File: app/schemas/reports.py
"""审核报告模型。"""

from datetime import UTC, datetime
from enum import StrEnum
from typing import Any
from uuid import UUID, uuid4

from pydantic import BaseModel, Field


class ReportStatus(StrEnum):
    """报告状态。"""

    DRAFT = "draft"
    SUCCEEDED = "succeeded"
    FAILED = "failed"


class ReviewReport(BaseModel):
    """绑定单据版本的不可变报告。"""

    report_id: UUID = Field(default_factory=uuid4)
    document_id: UUID | None = None
    document_version_id: UUID
    version_no: int = 1
    status: ReportStatus = ReportStatus.DRAFT
    report_status: ReportStatus | None = None
    overall_risk_level: str | None = None
    rule_version: str = ""
    content: str | dict[str, Any]
    generated_at: datetime | None = None
    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))


class ReportListItem(BaseModel):
    """报告列表摘要。"""

    report_id: UUID
    document_version_id: UUID
    status: ReportStatus
    report_status: ReportStatus | None = None
    overall_risk_level: str | None = None
    created_at: datetime


class ExportTaskResponse(BaseModel):
    """报告异步导出任务状态。"""

    export_task_id: UUID
    status: str
    file_name: str | None = None
    error_message: str | None = None


class ReportExportCommand(BaseModel):
    """报告导出请求。"""

    format: str = "pdf"

File: front/src/api/reports.ts
import type { ReviewReport } from '@/types/domain'
import { apiDownload, apiFetch } from './client'

/** 权威报告接口：按 document_version_id 读取（SPEC 15） */
export function getReviewReport(documentVersionId: string): Promise<ReviewReport> {
  return apiFetch<ReviewReport>(`/review-reports/${documentVersionId}`)
}

export interface ReportVersionRow {
  document_version_id: string
  version_no: number
  report_status: string
  overall_risk_level?: string
  generated_at?: string
}

export function listReportVersions(documentId: string): Promise<ReportVersionRow[]> {
  return apiFetch<ReportVersionRow[]>(`/documents/${documentId}/review-reports`)
}

export interface ExportTask {
  export_task_id: string
  status: 'pending' | 'running' | 'succeeded' | 'failed'
  file_name?: string
  error_message?: string
}

export function exportReviewReport(
  documentVersionId: string,
  format: 'pdf' | 'xlsx',
): Promise<ExportTask> {
  return apiFetch<ExportTask>(`/review-reports/${documentVersionId}/export`, {
    method: 'POST',
    body: { format },
    idempotencyKey: crypto.randomUUID(),
  })
}

export function getExportTask(exportTaskId: string): Promise<ExportTask> {
  return apiFetch<ExportTask>(`/review-reports/exports/${exportTaskId}`)
}

export function downloadExport(exportTaskId: string, fileName: string): Promise<void> {
  return apiDownload(`/review-reports/exports/${exportTaskId}/download`, fileName)
}
